app/src/org/commcare/android/storage/framework/Persisted.java

```diff
@@ -98,6 +97,7 @@ public int getID() {
     }
 
     private void readVal(Field f, Object o, DataInputStream in) throws DeserializationException, IOException, IllegalAccessException {
+        synchronized (fieldOrderings) {
```

Comment:
You should be able to just synchronize 'f' here instead, which should protect any of these methods which are trying to manipulate that particular field while still allowing others to be read/written.